### PR TITLE
Apply Common Actions guidance across Polaris Storybook and Doc site

### PR DESCRIPTION
**Pull request by aaronccasanova**
### WHY are these changes introduced?

Closes https://github.com/Shopify/polaris-internal/issues/1514

### WHAT is this pull request doing?

This PR applies the Common Actions guidance across Polaris Storybook and Doc site.

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#install-dependencies-and-build-workspaces)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [ ] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by alex-page — APPROVED** [review]

File: polaris-react/src/components/Tabs/Tabs.tsx [hunk 1]
```diff
@@ -524,7 +524,12 @@ export const Tabs = ({
           });
         }
       }}
-      icon={<Icon source={PlusIcon} accessibilityLabel={createViewA11yLabel} />}
+      icon={
+        <Icon
+          source={PlusCircleIcon}
```
Comment (aaronccasanova) [inline polaris-react/src/components/Tabs/Tabs.tsx]: cc @sarahill I had to apply this change directly to the `Tabs` component. Just want to confirm we should be using `PlusCircleIcon` in all cases?

**Review by sam-b-rose — APPROVED** [review]
Nice work, @aaronccasanova 🎉

**Review by lgriffee — APPROVED** [review]
✨

**Comment by sarahill** [comment]
@aaronccasanova  noticed some button examples need to be updated.

Update all critical examples with icons to use the delete icon. Looks like the secondary critical was updated but no the others.

Screenshot 2024-03-12 at 11 25 41 AM

**Review by sarahill — APPROVED** [review]

File: polaris-react/src/components/Tabs/Tabs.tsx (hunk 1 above)
Comment (sarahill) [reply polaris-react/src/components/Tabs/Tabs.tsx]: Ah I was wondering about that. Maybe lets hold off on that and make a separate issue. I want to get sign off before making the change to the component.

Comment (aaronccasanova) [reply polaris-react/src/components/Tabs/Tabs.tsx]: Sounds good 👍  https://github.com/Shopify/polaris/pull/11717/commits/ceb07d6540556cd965f4d469418158290e3cd99b

**Pull request merged**